cmd/zoekt-sourcegraph-indexserver/fetcher.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/google/zoekt"
+	"io"
+	"os/exec"
+	"strings"
+)
+
+// a deltaFetcher returns a list of changed files along with their file contents
+//
+// NOTE: The idea of returning the file contents in the same function seems awkward
+// with local checkouts, but necessary in a remote one
+type deltaFetcher interface {
+	// Fetch returns the paths that have changed on the latest
+	// commit for each branch, along with their file contents.
+	Fetch(ctx context.Context, branches ...zoekt.RepositoryBranch) (branchDeltas, error)
+}
+
+type branchDeltas map[string]delta
+
+// delta contains all the paths that have been added, changed, and removed
+// along with the file contents
+//
+// TODO@ggilmore: should the file contents  be io.ReadClosers, and should delta offer a "close"
+// method that closes all the file contents (thinking about the local file case)
+type delta struct {
+	// Added is a list of all the paths that have been added,
+	// along with their file contents.
+	Added map[string]io.Reader
```

Comment:
returning an io.Reader in a datastructure is a bit of a footgun since it is stateful.

Maybe we should have a tarball or a zipfile for all "modified" content. I suppose the consumer of this interface would enjoy having a map, not sure. The zipfile/tarball interface is simpler though and more direct. It's also how we would likely transport the files from gitserver.